docs/tutorials/local_iteration_series/testing_with_artificial_conditions.md

```diff
@@ -0,0 +1,173 @@
+	- [Debug Builds](#debug-builds)
+- [System-wide network conditioners](#system-wide-network-conditioners)
+	- [Clumsy on Windows](#clumsy-on-windows)
+		- [Settings quickstart](#settings-quickstart)
+	- [Network Link Conditioner on OSX](#network-link-conditioner-on-osx)
+		- [Settings quickstart](#settings-quickstart-1)
+
+
+When we are developing a multiplayer game that is intended to operate over the internet we inevitably have to deal with the reality of poor network conditions.
+
+Adverse network factors, such as [latency, jitter and packet loss](lagandpacketloss.md) all have to be taken into account during development and the tolerance thresholds for these should not be an afterthought or something relegated to an "optimization pass". 
+
+:::warning
+
+Not testing with artificial network conditions makes it significantly more likely to have unexpected behaviors when the game is running over the internet.
+
+:::
+
+This is challenging when we are [iterating on our game locally](local-Iteration-testing-multiplayer-games-locally.md) - after all, all of the game instances are still running on the same network interface. It's reasonable to expect that there will be little to no latency between the clients, which is no good for our purposes - we want the network to **misbehave**. 
+
+Thankfully there are a number of tools that can simulate adverse network conditions.
+
+For testing locally within the editor we suggest using [Unity Transport - Simulator Tools](#unity-transport---simulator-tools) along with [clone-based workflow via ParrelSync](testing_locally.md#parrelsync).
+
+For testing development builds with built-in artificial latency we suggest using [Unity Transport Simulator Tools with some custom code to inject artifical conditions into the build](#debug-builds).
+
+For testing release builds we suggest using either [Clumsy](#clumsy-on-windows) if you're on Windows, and [Network Link Conditioner](#network-link-conditioner-on-osx) if you're on Mac OS.
+
+## General advice
+
+It's not immediately obvious what the values and options we should enable in our network conditioning tools. All of them allow us to alter various aspects of network conditions such as latency, jitter and packet loss, though the names for concepts and the specific functionality varies between tools.
+
+The values we would use would be different depending on the exact debugging goal, but generally we want our artificial conditions to somewhat match up to what our users would be dealing with in reality.
+
+So how do we reason about what our lag threshold is? 
+To answer this question we can look for information regarding typical and acceptable delays for our game genre (Fighting games, First Person Shooters etc are on one side of the spectrum - these games require fast reaction times and as such have less of an acceptable threshold for lag, whereas the Realtime Strategy games are on the other side of the spectrum, generally being tolerant of much higher delays).
```

Comment:
That part I'm not sure about. Users won't switch to a better internet to play a different game genre. They'll have their home internet and that's it, the lag you use to test shouldn't depend on the game type.
If I'm at home and I play an FPS or a card game, in both cases, my ping will be the same.
Per platform lag, that I agree. In past projects, I've seen a clear difference in our live analytics between mobile vs desktop/consoles (and even between iOS and Android).
So I would test with higher lag if I want to target mobile for example, since there's a higher chances live users will have higher latencies.

Reply:
Em, im not sure, I switched my internet provide to get better internet. im pretty sure a lot of gamers when they move house check the provider speeds(I know I do). Users will go to the game that requires the best and work from there so in your case I would work out what i need for the FLS and switch to that, knowing the other games will work fine on it.

Reply:
I'd expect a games studio to have a target (or idea) for lag tolerance across all the targeted platforms and test against that. i.e. just getting a handle on how unplayable the game could be on x platform with severe lag. 
As developers, we have zero control over the player's internet speeds and internet hardware setup. 

It might not be good for business to say - _please upgrade your internet to Starlink for optimal gameplay experiences_

Reply:
I agree we can't say upgrade your internet, but we already have a section that outlines examples of how lag impacts different game types, this section seems to fit in with that general guidance. So explaining that if your designing a game that is turn based  lag might not be as impactful as it is on a FPS so testing with artificially high lag in the latter would be more useful than the former. So for instance Overwatch mIn specs states you need a broadband connection. so dialup wont cut it.

https://eu.battle.net/support/en/article/65159

Reply:
Oh, so this bit is not about the user and their internet - it's about answering a question "what is the minimal latency I should even be testing with, and what is the typical latency under which other games in the same genre tend to fare well". It's ta gauge reasonable expectations for your own code - under what conditions should it perform adequately.

Reply:
As in, the team needs to define that "our game should be playable within 0 to 150ms lag with no perceptible degradation of experience, 150-200 ms is bad, but it could still work, everything else we don't care about"

Reply:
Yeah, I think we are eluding to the same topic, so maybe the feedback here is go a bit wider with information, as it could be misinterpreted or lacking details?

Reply:
Rephrased this section a bit.

Reply:
Overwatch fails elegantly above 300ms. In the sense where you can still play the game, but server rewind will stop helping you after 300ms and so you'll start missing your shots more. This is a choice Blizzard made. Other FPS devs will not put such a restriction and will allow 500ms and even 1000ms latency.
In past projects, our QA team would test with 100ms latency, 200ms, 500ms and 1000ms and give a report on playability. It'd then be up to designers/leads to determine if the issues found were acceptable or not, if they were ship blockers or not.
Determining which network conditions to test with is NOT about what type of game you have. It's about what your users will encounter while playing your game, which will vary by region and by platform, not by genre.
Saying an FPS will require lower latencies doesn't make sense --> tell that to your roommate trying to download game of thrones while you play. We have lag hiding techniques for this, we have prediction, we have server rewind, we have interpolation/buffering to hide that jitter, but that 200ms of lag won't disappear, it's completely our of your control as a player and as a dev. It's up to the dev to make sure it doesn't show. And to do that, you need to test with realistic network conditions and use the right techniques to hide it successfully.

Now determining what bugs are ACCEPTABLE after you tests latency, that's dependent on game genre. If I have a turn by turn card game and I don't get an update after 500ms due to latency, I wouldn't mind. Or if my tank is moving slow because I'm not predicting it, I wouldn't mind (I'd call this a feature and say it's due to the big thing's intertia... :P ) But you should still test at 100ms, 200ms, 500ms, 1000ms, to make sure you don't have weird race conditions, since your users WILL have those latencies.

Again, test values for artificial network conditions are orthogonal to game genre.

Reply:
I'll add
*what's acceptable is dependant on game genre AND on your game design team and leads. Market analysts could show you should target users from countries with higher latencies for example. It's up to you to determine what's acceptable, there's no single rule for this.
but you should still test different latencies!!

Reply:
Brian your link on Overwatch requirements specifies bandwidth requirements, not latency requirements. They are very different. You can have good bandwidth and very bad latency. Notice how they don't specify required ping to play

Reply:
Thank you Sam, you're 100% right. I've reworded the segment to reflect ^^.